### PR TITLE
Delete stream destination filters when deleting a stream

### PR DESCRIPTION
**Pull request by patrickmann**
/nocl unreleased
Resolves Graylog2/graylog-plugin-enterprise#7833 

Add handler for stream deletion to `StreamDestinationFilterService`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java [hunk 1]
```diff
@@ -175,4 +183,13 @@ public void forEachEnabledFilterGroupedByStream(Consumer<GroupByStreamResult> co
                 ))
         ), GroupByStreamResult.class).forEach(consumer);
     }
+
+    @Subscribe
+    @SuppressWarnings("unused")
+    public void handleStreamDeleted(StreamDeletedEvent streamDeletedEvent) {
```
Comment (kodjo-anipah) [inline graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java:189]: Can we directly just deleted it? That shouldn't be coming from an api call, so I don't think there is any need to fetch the filter, verify if the filter matches the streamId before deleting it 
````
    public void handleStreamDeleted(StreamDeletedEvent streamDeletedEvent) {
        collection.deleteOne(Filters.eq(FIELD_STREAM_ID, streamDeletedEvent.streamId()));

    }
    
    ````

Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java:187]: Good point. Except, are we sure there is only a single rule per stream? I think we should call `deleteMany`.

Comment (kodjo-anipah) [reply graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java:187]: oh right, you are absolutely right, we should delete all rules

**Review by kodjo-anipah — APPROVED** [review]
LGTM

**Pull request merged**